### PR TITLE
Update OCCT and pythonocc to 7.6.2 

### LINKED ISSUE #31: Update pythonocc to 7.6.2

### PR DESCRIPTION
**Pull request by joergbrech**
Closes #31

### CODE REVIEW

**Comment by AntonReiswich** [comment]
The last action required here is the conflict. I assume in both conflicts the number 0 is to keep.

**Comment by joergbrech** [comment]
Please merge this after #35

**Comment by joergbrech** [comment]
@AntonReiswich This is ready for review again. Note that I had an issue after the merge of #35:

 - The github runner image `ubuntu-20.04` comes with swig 4.0.1 pre-installed, while `ubunut-22.04` has swig 4.0.2 pre-installed, see the readmes [here](https://github.com/actions/runner-images/tree/main/images/ubuntu).
 - building pythonocc now requires `swig>=4.0.2`. This requirement is added to the `meta.yml` file and `conda-build` installs the correct swig version into a dedicated build environment. 
 - For some reason, the system-wide installation `/usr/bin/swig4.0` gets picked up by cmake during the `conda-build` command instead of the one from the decicated build environment. This wasn't an issue on `ubuntu-22.04`, where the correct version was installed, but it is on `ubuntu-20.04`: Cmake complains that it finds an unsuitable swig version, ignoring the one from the conda-build build environment.
 - As a workaround I just `sudo rm /usr/bin/swig4.0` on `ubuntu-20.04` as part of the CI.

File: opencascade/fix-private-linking.patch [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-+++ opencascade-7.4.0/adm/cmake/occt_toolkit.cmake
-@@ -354,7 +354,7 @@ else()
- endif()
- 
- if (BUILD_SHARED_LIBS)
+--- opencascade-7.6.2.orig/adm/cmake/occt_toolkit.cmake
++++ opencascade-7.6.2/adm/cmake/occt_toolkit.cmake
+@@ -400,7 +400,7 @@
+   if(IS_VTK_9XX)
+     string (REGEX REPLACE "vtk" "VTK::" USED_TOOLKITS_BY_CURRENT_PROJECT "${USED_TOOLKITS_BY_CURRENT_PROJECT}")
```
Comment (AntonReiswich) [inline opencascade/fix-private-linking.patch:7]: Hm, I don't understand what this conditional and this line is for??

Comment (joergbrech) [reply opencascade/fix-private-linking.patch:7]: I didn't write this patch manually, but created it using `git diff`. The patch file contains the lines *preceeding* the actually modified lines, which are given in line 9 and 10 below. The patch makes sure that transient dependencies are linked privately.

So this conditional wasn't written by me, but is part of opencascade.

Comment (AntonReiswich) [reply opencascade/fix-private-linking.patch:7]: I see, thank you!

**Pull request merged**